### ISSUE TITLE
Port numbers are lost on standard schemes.

### ISSUE THREAD

**Issue by drivron** [electron/electron#24725]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v10.0.0-beta.12
* **Operating System:**
  * Ubuntu 20.04 x64

### Expected Behavior
The port number is specified in the RFC 3986 and should be included in the URL.

### Actual Behavior
The request URL in a standard protocol handler does not contain the port number.

### To Reproduce
This electron fiddle shows the difference between a standard and a non standard protocol :
https://gist.github.com/drivron/3e8fdb461f78c3b926571836fd14ad41

**Comment by RangerMauve** [comment]
Is there something I can do to fix this? Would it be possible to get a pointer to where this URL parsing is happening? Would a PR that fixes it be welcome?

**Comment by RangerMauve** [comment]
I think all that's needed is to change this line:

https://github.com/electron/electron/blob/master/shell/browser/api/electron_api_protocol.cc#L106

from `url::SCHEME_WITH_HOST` to `url::SCHEME_WITH_HOST_PORT_AND_USER_INFORMATION`

Seems like a pretty simple fix.

**Comment by RangerMauve** [comment]
Took another crack at this. 😁

**Comment by drivron** [comment]
This issue is still present in Electron 21.1.0.

**Comment by RangerMauve** [comment]
Yeah, sadly I haven't had time to work on the fix I attemted since I've been busy with work. 😩

If anyone else can take a look, I think I was on the right track.

**Comment by RangerMauve** [comment]
This is still important, I just haven't had time to work on it myself and am still figuring out funding to hire somebody to work on it.

**Comment by RangerMauve** [comment]
This is still an issue! Sadly I still haven't had time to work on it more in my spare time and haven't found any clients willing to fork out the dev hours. :P

**Comment by RangerMauve** [comment]
I could probably negotiate a bounty for someone that's willing to tackle it though?

**Comment by drivron** [comment]
This issue is still present in Electron 26.0.0-beta.4. Using the new protocol.handle method does not change the problem.

**Comment by drivron** [comment]
bump

**Comment by RangerMauve** [comment]
This is still something I'd like. Sadly I have had no spare time for it TT_TT

**Comment by BurningEnlightenment** [comment]
bump

**Comment by BurningEnlightenment** [comment]
🪄 unstale

**Comment by BurningEnlightenment** [comment]
🪄 unstale

**Comment by BurningEnlightenment** [comment]
🪄 unstale

**Comment by drivron** [comment]
bump

**Comment by RangerMauve** [comment]
I wish I had time to follow up on this but if someone wants I can pair with them for an hour or two to talk about what I had gotten working so far. It was just failing some tests but I think we're close to something working

**Comment by RangerMauve** [comment]
Still relevant!

**Comment by BurningEnlightenment** [comment]
🪄 unstale

**Comment by BurningEnlightenment** [comment]
🪄 unstale

**Comment by RangerMauve** [comment]
Still relevant sadly

**Comment by BurningEnlightenment** [comment]
🪄 unstale
